runtime/util/vmargs.c

```diff
@@ -506,22 +510,28 @@ optionValueOperations(J9PortLibrary *portLibrary, J9VMInitArgs* j9vm_args, IDATA
 				done = TRUE;
 				break;
 
+			case GET_DBL_VALUE:
+				*(double *)reserved = 0.0;
+				cursor = getOptionValue(j9vm_args, element, *valuesBuffer);
+				rc = scan_double(&cursor, &dvalue);
+				if (*cursor != '\0') {
+					return OPTION_MALFORMED;
+				}
+				if (2 == rc) {
+					return OPTION_OVERFLOW;
+				} else if (1 == rc) {
+					return OPTION_MALFORMED;
+				}
+				*(double *)reserved = dvalue;
+				done = TRUE;
+				break;
+
 			case GET_MEM_VALUE :
 			case GET_INT_VALUE :
 			case GET_PRC_VALUE :
-				*((UDATA*)(reserved)) = 0;				/* Initialize the return value */
-				if ( HAS_MAPPING(j9vm_args, element) ) {
-					/* If we have a mapping, then we ignore the option name passed in. Instead we want the one that corresponds to the actual cmd line */
-					scanStart = &(j9vm_args->actualVMArgs->options[element].optionString[ strlen(MAPPING_MAPNAME(j9vm_args, element)) ]);
-				} else {
-					if (*valuesBuffer) {
-						scanStart = &(j9vm_args->actualVMArgs->options[element].optionString[ strlen(*valuesBuffer) ]);
-					} else {
-						return OPTION_ERROR;
```

Comment:
For reviewers: this "else" is not necessary because of check lines 363-364, it is removed in new code